src/components/Article.jsx

```diff
@@ -57,7 +63,52 @@ export default function Article() {
 
     if(hasError) return <section className="page-not-found"></section>
 
-return <div className="article-container">
+    const articleVoteAdjustment = (buttonPressed, alreadyVoted) => {
+        let adjust = 1
+        if(buttonPressed === "dislike-button") adjust = -1;
+        if(alreadyVoted) {
+            adjust = -1
+            if(buttonPressed === "dislike-button") adjust = 1;
+        }
+        if(articleVotedOn && buttonPressed !== articleVotedOn) {
+            const oppositeButton = document.getElementById(articleVotedOn)
+            oppositeButton.classList.remove("voted")
+            adjust = adjust === 1 ? 2 : -2
+        }
+        return adjust
+    }
+
+    const likeHandler = (e) => {
+        setArticleVoteError(false)
+        if(!user.username) {
+            setArticleVoteError("Please sign in to vote")
+            return
+        }
+        const buttonPressed = e.target.classList[0]
+        const alreadyVoted = e.target.classList[1]
+        const adjust = articleVoteAdjustment(buttonPressed, alreadyVoted)
+        setArticleVotes(articleVotes + adjust)
+        patchArticles(`${article_id}`,{ inc_votes: adjust })
+        .then(() => {
+            setArticleVoteError(false)
+            if(alreadyVoted) {
+                e.target.classList.remove("voted")
+                setArticleVotedOn(false)
+            }
+            else {
+                e.target.classList.add("voted")
+                setArticleVotedOn(buttonPressed)
+            }
+        })
```

Comment:
This is a good start, but on a slow connection this doesnt work as well. This is because the idea of updated the class like this is delayed until the response has been received, instead I would recommend looking into optimistic rendering to this behaviour.

Notes on optimistic rendering:
https://l2c.northcoders.com/courses/sd-notes/front-end#sectionId=optimistic-rendering,step=intro